.github/workflows/python-test.yml

```diff
@@ -0,0 +1,144 @@
+name: Run tests
+
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+  schedule:
+    - cron: "0 0 * * *"
+
+permissions:
+  contents: read
+
+jobs:
+
+  test-ubuntu:
+
+    runs-on: ubuntu-latest
+
+    steps:
+    - uses: actions/checkout@v4
+
+    - name: Set up Python 3.10
+      uses: actions/setup-python@v3
+      with:
+        cache: 'pip'
+        python-version: '3.10'
+
+    - name: Install dependencies
+      run: |
+        sudo apt-add-repository ppa:ubuntugis/ubuntugis-unstable
```

Comment:
Do we want to always test against unstable? It sounds like this would be better tested in a `matrix` configuration. https://docs.github.com/en/actions/using-jobs/using-a-matrix-for-your-jobs

Reply:
This step is not actually selecting the OS version, but rather installing the necessary system binaries for GDAL, which is required for some Python GIS libraries. But when you pointed this out I noticed that it would indeed be better to use the stable ppa, so I updated it.

I think that it is not necessary to test on different OS versions, since this library mostly does just handle a lot of REST requests. However, I liked the idea of using a strategy matrix to test against different Python verisons, so I made it do that.

Thoughts? Counterpoints?

Reply:
@iboates Please excuse the delay, I've been out of the office for the last week!

I agree with you on the OS point; Unless we're testing things that are platform-dependent, we don't really need to test different OSes. Typically, if a package claims to support a Python version, we should be testing said version via a matrix to be sure it's supported (sometimes dependencies drop the lowest version without any announcement, etc.). Changes here look good to me!